iceprod/server/data/www/rest.js

```diff
@@ -0,0 +1,51 @@
+/*!
+ * Helper functions that use REST API
+ */
+
+async function fetch_json(url, passkey, json=null) {
+	try {
+		var payload = {headers: new Headers({'Authorization': passkey})};
+		if (json) {
+			payload['body'] = JSON.stringify(json);
+			payload['method'] = 'PUT';
+		}
+		var response = await fetch(url, payload);
+		return await response.json();
+	} catch(err) {
+		alert(err);
+	}
+}
+
+function set_dataset_status(dataset_id, stat, passkey, task_filters=[''], propagate=true) {
+	if (propagate) {
+		let jobs = fetch_json('/datasets/' + dataset_id + '/jobs', passkey);
```

Comment:
I'm not sure if this line will actually return the json object, or if it will return a Promise for that object.

Reply:
I think the actual object is returned because fetch_json uses await: `return await response.json();`, but I don't know.